### ISSUE TITLE
Docs are wrong for Typescript projects

### ISSUE THREAD

**Issue by lud-hu** [histoire-dev/histoire#669]
### Describe the bug

For TS + nuxt projects, I cannot just do what the docs say about adding the `env.d.ts` into the tsconfig. It will just overwrite the nuxt config and therefore vanish out all auto imports etc. If I copy over the generated tsconfig from the `.nuxt` directory it's working, but this is more like a workaround.

### Reproduction

https://github.com/lud-hu/nuxt-histoire-reproduction/blob/main/tsconfig.json

### System Info

```shell
System:
    OS: macOS 14.0
    CPU: (8) arm64 Apple M1
    Memory: 68.38 MB / 16.00 GB
    Shell: 5.9 - /bin/zsh
  Binaries:
    Node: 18.19.0 - /opt/homebrew/opt/node@18/bin/node
    Yarn: 1.22.19 - /opt/homebrew/bin/yarn
    npm: 10.2.3 - /opt/homebrew/opt/node@18/bin/npm
    pnpm: 8.15.1 - /opt/homebrew/bin/pnpm
  Browsers:
    Chrome: 121.0.6167.139
    Edge: 121.0.2277.98
    Safari: 17.0
  npmPackages:
    @histoire/plugin-nuxt: ^0.17.9 => 0.17.9 
    @histoire/plugin-vue: ^0.17.9 => 0.17.9 
    histoire: ^0.17.9 => 0.17.9
```

### Used Package Manager

yarn

### Validations

- [X] Follow our [Code of Conduct](https://github.com/histoire-dev/histoire/blob/main/CODE_OF_CONDUCT.md)
- [X] Read the [Contributing Guidelines](https://github.com/histoire-dev/histoire/blob/main/CONTRIBUTING.md).
- [X] Read the [docs](https://histoire.dev/guide/).
- [X] Check that there isn't [already an issue](https://github.com/histoire-dev/histoire/issues) that reports the same bug to avoid creating a duplicate.
- [X] Check that this is a concrete bug. For Q&A open a [GitHub Discussion](https://github.com/histoire-dev/histoire/discussions).
- [X] The provided reproduction is a [minimal reproducible example](https://stackoverflow.com/help/minimal-reproducible-example) of the bug.

**Comment by letoast** [comment]
Same thing here. I managed to solve it by importing the path in `nuxt.config.ts`

```ts
export default defineNuxtConfig({
	imports: {
		imports: [
			{
				from: './node_modules/@histoire/plugin-vue/components.d.ts',
				type: true,
			},
		],
	},
})
```

**Comment by YaredFall** [comment]
Just added `.d.ts` at the end of types reference path in env.d.ts and its worked!

```(ts)
/// <reference types="./node_modules/@histoire/plugin-vue/components.d.ts" />
```

**Comment by letoast** [comment]
> Just added `.d.ts` at the end of types reference path in env.d.ts and its worked!
> 
> ```
> /// <reference types="./node_modules/@histoire/plugin-vue/components.d.ts" />
> ```

You get global components typed or do you have to import them? Because what I wrote in my reply just does what the docs say should be done, it doesn't solve the problem for me, the components are still undefined. In order for it to work @histoire would need to add to here: https://github.com/histoire-dev/histoire/blob/main/packages/histoire-plugin-vue/components.d.ts#L217

```ts
module '@vue/runtime-core' {
  export interface GlobalComponents {
    ...whichever components need to be autoimported...
  }
}

module '@vue/runtime-dom' {
   export interface GlobalComponents {
    ...whichever components need to be autoimported...
  }
}
```

If I do this and modify it in node_modules then it actually works. But this shouldn't be modified in node_modules, it should be fixed by the package authors.

**Comment by YaredFall** [comment]
@letoast Yes, I get global types for histoire components without the need to import them.
I am using Nuxt too.

I have created a [repo](https://github.com/YaredFall/Nuxt-Histoire-starter) with fresh nuxt and histoire setup. Histoire components globally available and have proper types (at least for me).

**Comment by letoast** [comment]
@YaredFall looks like there was something in my project (it's a pretty big project with lots of layers). I had to remove node_modules/, .nuxt/ folder and yarn.lock file and reinstall everything again. Now it works. Thanks for the repo, it made me look for the issue again :D
